packages/amplify-provider-awscloudformation/src/utils/consolidate-apigw-policies.ts

```diff
@@ -246,8 +246,14 @@ function updateExistingApiCfn(context: $TSContext, api: $TSObject): void {
   const apiParamsFile = path.join(resourceDir, API_PARAMS_FILE);
   const cfn: any = JSONUtilities.readJson(cfnTemplate, { throwIfNotExist: false }) ?? {};
   const parameterJson = JSONUtilities.readJson(paramsFile, { throwIfNotExist: false }) ?? {};
-  const parameters = cfn?.Parameters ?? {};
-  const resources = cfn?.Resources ?? {};
+
+  if (!cfn) {
+    // This should not happen. If it does, nothing else will work anyway so just bail out.
```

Comment:
Can't we just throw an error to make sure a return here will not makes things bad? We have a similar throw for layers when a value MUST be present that further functionality depends on.

Reply:
Sure, I can update it to throw.

Reply:
Updated!

Reply:
I think throwing returning would make the CLI proceed with pushing the category